examples/customOpenAPIschema.md

```diff
@@ -0,0 +1,230 @@
+# Using a Custom OpenAPI schema
+
+A kustomization file supports adding your own
+OpenAPI schema to define merge keys and patch
+strategy.
+
+Make a place to work:
+
+<!-- @placeToWork @testAgainstLatestRelease -->
+```
+DEMO_HOME=$(mktemp -d)
+```
+
+We'll be editing our own CRD:
+
+<!-- @customOpenAPI @testAgainstLatestRelease -->
+```
+cat <<EOF >$DEMO_HOME/mycrd.yaml
+apiVersion: example.com/v1alpha1
+kind: MyCRD
+metadata:
+  name: service
+spec:
+  template:
+    spec:
+      containers:
+      - name: server
+        image: server
+        command: example
+        ports:
+        - name: grpc
+          protocol: TCP
+          containerPort: 8080
+EOF
+```
+
+The edit we want to make is change the image to nginx. We can
+do that via a strategic merge patch. 
```

Comment:
This resource has an `image` field.  Let's change it's value from `server` to `nginx` with a patch.

Reply:
changed